### ISSUE TITLE
peak detector: minimum datapoints

### ISSUE THREAD

**Issue by stvnrlly** [cantino/huginn#801]
In some cases, I'd like to let the Peak Detector Agent gather data for a little while before attempting to identify peaks. This would be useful when the agent is doing something like powering a Twitter account, to avoid early false peak reports.

**Comment by virtadpt** [comment]
Hear, hear!  Oh gods, hear hear...

**Comment by stvnrlly** [comment]
This might be as easy as allowing an option to set the minimum value here: https://github.com/cantino/huginn/blob/master/app/models/agents/peak_detector_agent.rb#L63.

**Comment by stvnrlly** [comment]
It might also be possible to distinguish between high-volume streams where gathering more data before detecting is useful and low-volume streams where smaller variations are more noteworthy. This could be done by calculating how long ago the 4th-most-recent event occurred before determining whether to use the high or low minimum.

**Comment by stvnrlly** [comment]
I actually have this coded, but just did a bad job of putting it on its own branch. Once #808 is merged, I'll package that code into a new pull request.

**Comment by Jngai** [comment]
@stvnrlly are you still working on this? I may be interested.

**Comment by stvnrlly** [comment]
@Jngai not currently, unfortunately. I had written some code, but did a bad job of actually committing it and now can't find it any more :(.

**Comment by Jngai** [comment]
Hey @dsander I am having trouble coming up with an example since I don't know any particular agent well enough to check its peak. Do you have an example for me? It should click when I see something.

**Comment by cantino** [comment]
I usually use it with a TwitterStreamAgent.  See http://blog.andrewcantino.com/blog/2014/03/17/know-when-the-world-changes-with-huginn/

**Comment by dsander** [comment]
@Jngai Me too, you could also feed it stock prices.
